File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        minLen = len(strs[0])
        
        for i in range(len(strs)):
            minLen = min(minLen, len(strs[i]))
            
        res = ''
        
        for i in range(minLen):
            s = strs[0][i]
            for j in range(len(strs)):
                if s != strs[j][i]:
                    return res
            res += s
        return res

File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        int len = strs.length;
        if(len == 0){
            return "";
        }

        String prefix = strs[0];

        for(int i=1; i<len;i++){
            while(strs[i].indexOf(prefix)!=0){
                prefix = prefix.substring(0, prefix.length()-1);
            }
        }
        return prefix;

        
    }
}